pkg/kubectl/cmd/create.go

```diff
@@ -75,19 +75,24 @@ func RunCreate(f *cmdutil.Factory, out io.Writer, filenames util.StringList) err
 		return err
 	}
 
-	cmdNamespace, err := f.DefaultNamespace()
+	cmdNamespace, overridden, err := f.DefaultNamespace()
 	if err != nil {
 		return err
 	}
 
 	mapper, typer := f.Object()
-	r := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
+	b := resource.NewBuilder(mapper, typer, f.ClientMapperForCommand()).
 		Schema(schema).
 		ContinueOnError().
-		NamespaceParam(cmdNamespace).RequireNamespace().
+		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(filenames...).
-		Flatten().
-		Do()
+		Flatten()
+
+	if overridden {
```

Comment:
Since this is specifically related to the spec files, would it be better to change FilenameParam to accept a leading bool for this? This way if you ever use the builder with spec files, you have to consider this behavior.